Products/ZenModel/MaintenanceWindow.py

```diff
@@ -555,6 +555,11 @@ def setProdState(self, state, ending=False, batchSize=None,
 
         def _setProdState(devices_batch):
             for device in devices_batch:
+                # In case we try to move Component Group into MW
+                # some of objects in CG may not have production state
+                # we skip them.
+                if not hasattr(device, "privateattr_productionState"):
```

Comment:
We should not reference privateattr_productionState directly.  Also, what are the cases where this is true?  Most components will NOT have a production state, but will acquire their device's production state.